internals/overlord/logstate/clienterr/err.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package clienterr
+
+import (
+	"time"
+)
+
+// Backoff should be returned if the server indicates we are sending too many
+// requests (e.g. an HTTP 429 response).
+type Backoff struct {
+	RetryAfter *time.Time
+}
+
+func (b Backoff) Error() string {
+	errStr := "too many requests"
+	if b.RetryAfter != nil {
+		errStr += ", retry after " + b.RetryAfter.String()
+	}
+	return errStr
+}
+
+// Timeout should be returned if the client's Flush times out.
+type Timeout struct{}
```

Comment:
it could, but I'd rather use the same pattern for all errors in this package. It's confusing if we have to handle `clienterr.Timeout` differently to `clienterr.Backoff`.

Reply:
plus, we might want to add more context / data / information to this error in the future